changes/views.py

```diff
@@ -72,3 +78,54 @@ def post(self, request, *args, **kwargs):
                               action='momconnect_nonloss_optout',
                               data={'reason': 'sms_failure'})
         return Response(status=status.HTTP_201_CREATED)
+
+
+class ReceiveAdminOptout(mixins.CreateModelMixin,
+                         generics.GenericAPIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+
+        try:
+            data = utils.json_decode(request.body)
+        except ValueError as e:
+            return JsonResponse({'reason': "JSON decode error",
+                                'details': six.text_type(e)}, status=400)
+
+        try:
+            identity_id = data['identity']
+        except KeyError as e:
+            return JsonResponse({'reason': '"identity" must be specified.'},
+                                status=400)
```

Comment:
Same comment here as in https://github.com/praekelt/hellomama-registration/pull/139/ about rather user a serializer for this validation.